SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -1010,6 +1010,23 @@ void USpatialReceiver::ApplyComponentUpdate(const Worker_ComponentUpdate& Compon
 	ComponentReader Reader(NetDriver, ObjectReferencesMap, UnresolvedRefs);
 	Reader.ApplyComponentUpdate(ComponentUpdate, TargetObject, Channel, bIsHandover);
 
+	// This is a temporary workaround, see UNR-841:
+	// If the update includes tearoff, close the channel and delete the entity.
+	if (TargetObject->IsA<AActor>() && ClassInfoManager->GetCategoryByComponentId(ComponentUpdate.component_id) == SCHEMA_Data)
+	{
+		TArray<uint32> UpdatedIds;
+		Schema_Object* ComponentObject = Schema_GetComponentUpdateFields(ComponentUpdate.schema_type);
+		UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
+		Schema_GetUniqueFieldIds(ComponentObject, UpdatedIds.GetData());
+
+		// The third Id is btearoff on an actor.
+		if (UpdatedIds.Contains(SpatialConstants::ACTOR_TEAROFF_ID))
```

Comment:
Do we care what the value is? Is there any reason we can't use Schema_GetBool?

Reply:
It may be safer to get bool - this runs under the assumption that we start off with tearoff=false, so the only update we would receive on that field would be tearoff=true. I added a check to see if GetBool ever returned false in this case, and it seemed not to.

Having more of a look at GetBool, it seems it returns the default value if the field doesn't exist, so I guess I could use that, and it'd be neater.